### PR TITLE
iR 594 Audit and add postprocessing effects

### PR DESCRIPTION
**Pull request by MbfloydIR**
Improve PostProcessing API

Added the missing post processing effects

GodRaysEffect needs light source that is not available, so it is left out

The following effects are aded but no difference can be seen when applied

SMAAEffect
SSAOEffect
SSREffect
DepthOfFieldEffect
BloomEffect
SSGIEffect
TRAAEffect
MotionBlurEffect
ShockWaveEffect

### CODE REVIEW

**Comment by DanielBelmes** [comment]
@MbfloydIR you have some errors on the checks here btw. Double check your commit hooks are running.

File: packages/editor/src/components/properties/PostProcessingSettingsEditor.tsx [hunk 1]
```diff
@@ -90,7 +97,7 @@ const EffectsOptions: Partial<EffectOptionsType> = {
     intensity: { propertyType: PropertyTypes.Number, name: 'Intensity', min: 0, max: 10, step: 0.01 },
     fade: { propertyType: PropertyTypes.Number, name: 'Fade', min: -1, max: 1, step: 0.01 },
     resolutionScale: { propertyType: PropertyTypes.Number, name: 'Resolution Scale', min: -10, max: 10, step: 0.01 },
-    kernelSize: { propertyType: PropertyTypes.Number, name: 'Kerne Size', min: 1, max: 5, step: 1 },
+    kernelSize: { propertyType: PropertyTypes.KernelSize, name: 'KernelSize' },
```
Comment (DanielBelmes) [inline packages/editor/src/components/properties/PostProcessingSettingsEditor.tsx]: The min max and step for this was slightly correct. I actually think there is a KernelSelect we could use for this with values from 0-5. There are others that can use it too.

File: packages/spatial/src/renderer/effects/PostProcessing.ts [hunk 2]
```diff
@@ -401,52 +443,52 @@ export type EffectPropsSchema = {
...
+  [Effects.FXAAEffect]: FXAAEffectProps
+  [Effects.TextureEffect]: TextureEffectProps
   [Effects.VignetteEffect]: VignetteEffectProps
-  // [Effects.LensDistortionEffect]: LensDistortionEffectProps
+  [Effects.LensDistortionEffect]: LensDistortionEffectProps
 }
 
 export type EffectPropsSchemaType = (typeof defaultPostProcessingSchema)[keyof typeof defaultPostProcessingSchema]
 
 export const defaultPostProcessingSchema: EffectPropsSchema = {
```
Comment (DanielBelmes) [inline packages/spatial/src/renderer/effects/PostProcessing.ts:415]: Were you able to check if the default values for the other effects matched the default values from the library/demos?

File: packages/spatial/src/renderer/functions/configureEffectComposer.ts [hunk 3]
```diff
@@ -142,27 +161,96 @@ export const configureEffectComposer = (entity: Entity, schema?: typeof defaultP
...
+      composer[key] = eff
+      effects.push(eff)
+    } else if (key == Effects.TextureEffect) {
+      let texturePath = effectOptions.texturePath
+      if (texturePath == undefined) {
+        texturePath = null
+      }
+      let texture: Texture | null = null
+      if (texturePath != null) {
+        let textLoad = new TextureLoader()
```
Comment (DanielBelmes) [inline packages/spatial/src/renderer/functions/configureEffectComposer.ts]: @MichaelEstes Do you have anyway with the resourceManager to load something out of a reactor?

File: packages/spatial/src/renderer/functions/configureEffectComposer.ts [hunk 4]
```diff
@@ -142,27 +161,96 @@ export const configureEffectComposer = (entity: Entity, schema?: typeof defaultP
...
+        //have to wait for the texture's image to load and then add the pass to the composer
+        texture = textLoad.load(texturePath, (textureLoaded) => {
+          effectOptions.texture = textureLoaded
+          const eff = new EffectClass(effectOptions)
+          composer[key] = eff
+          effects.push(eff)
+          AddPass()
```
Comment (DanielBelmes) [inline packages/spatial/src/renderer/functions/configureEffectComposer.ts]: I believe you're creating multiple duplicates of the EffectPass with `AddPass()`. We have to refactor this logic into PostProcessingComponent so we can use useEffects and the resource manager in order to load these textures.

**Review by DanielBelmes — CHANGES_REQUESTED** [review]
Some small stuff at first but I believe we do have to refactor configure Effect composer into the PostProcessingComponent earlier than expected for the texture effects.

File: packages/spatial/src/renderer/components/PostProcessingComponent.tsx [hunk 5]
```diff
@@ -63,10 +83,220 @@ export const PostProcessingComponent = defineComponent({
 
     useEffect(() => {
       if (!rendererEntity) return
-      configureEffectComposer(
-        rendererEntity,
-        postprocessingComponent.enabled.value ? postprocessingComponent.effects.get(NO_PROXY_STEALTH) : undefined
-      )
+      //configureEffectComposer(
```
Comment (HexaField) [inline packages/spatial/src/renderer/components/PostProcessingComponent.tsx]: we should definitely keep this as a separate function, long useEffects are hard to read

**Review by HexaField — CHANGES_REQUESTED** [review]

**Review by HexaField — CHANGES_REQUESTED** [review]
we should add a supreactor for each of the effects, we can then react to them individually, as well as simplify the logic around the texture loading

File: packages/spatial/src/renderer/components/PostProcessingComponent.tsx [hunk 6]
```diff
@@ -78,7 +78,7 @@ export const PostProcessingComponent = defineComponent({
   /** @todo this will be replaced with spatial queries or distance checks */
   reactor: () => {
     const entity = useEntityContext()
-    const rendererEntity = useScene(entity)
+    const rendererEntity = useScene(Engine.instance.viewerEntity)
```
Comment (HexaField) [inline packages/spatial/src/renderer/components/PostProcessingComponent.tsx]: why this change? this means any asset panels will not have postprocessing and apply it to the main canvas

**Review by HexaField — CHANGES_REQUESTED** [review]

**Comment by HexaField** [comment]
closing in favour of continued work https://github.com/EtherealEngine/etherealengine/pull/10148

**Pull request closed without merging**